app/src/leader.c

```diff
@@ -0,0 +1,362 @@
+                struct leader_seq_cfg *seq = completed_sequence_candidates[i];
+                if (seq->immediate_trigger || (num_candidates == 1 && num_comp_candidates == 1)) {
+                    press_leader_behavior(seq, data->timestamp);
+                }
+            }
+        } else { // keyup
+            if (data->position == active_leader_position && !first_release) {
+                first_release = true;
+                return 0;
+            }
+            if (!is_in_current_sequence(data->position)) {
+                return 0;
+            }
+            if (num_candidates == 0) {
+                zmk_leader_deactivate();
+                return ZMK_EV_EVENT_HANDLED;
+            }
+
+            release_count++;
+            release_key_in_sequence(data->position);
+
+            for (int i = 0; i < num_comp_candidates; i++) {
+                struct leader_seq_cfg *seq = completed_sequence_candidates[i];
+                if (seq->is_pressed && all_keys_released()) {
+                    release_leader_behavior(seq, data->timestamp);
+                    num_comp_candidates--;
+                }
+                if (num_candidates == 1 && num_comp_candidates == 0) {
+                    zmk_leader_deactivate();
+                }
+            }
+            if (!timerless || num_comp_candidates < num_candidates) {
+                reset_timer(data->timestamp);
+            }
+        }
+        return ZMK_EV_EVENT_HANDLED;
+    }
+
+    return 0;
+}
+
+ZMK_LISTENER(leader, position_state_changed_listener);
+ZMK_SUBSCRIPTION(leader, zmk_position_state_changed);
+
+#define LEADER_INST(n)                                                                             \
+    static struct leader_seq_cfg sequence_config_##n = {                                           \
+        .virtual_key_position = ZMK_KEYMAP_LEN + __COUNTER__,                                      \
+        .immediate_trigger = DT_PROP(n, immediate_trigger),                                        \
+        .is_pressed = false,                                                                       \
+        .key_positions = DT_PROP(n, key_positions),                                                \
+        .key_position_len = DT_PROP_LEN(n, key_positions),                                         \
+        .behavior = ZMK_KEYMAP_EXTRACT_BINDING(0, n),                                              \
+        .layers = DT_PROP(n, layers),                                                              \
+        .layers_len = DT_PROP_LEN(n, layers),                                                      \
+    };
+
+DT_INST_FOREACH_CHILD(0, LEADER_INST)
+
+#define INTITIALIAZE_LEADER_SEQUENCES(n) intitialiaze_leader_sequences(&sequence_config_##n);
+
+static int leader_init() {
+    k_work_init_delayable(&release_timer, behavior_leader_key_timer_handler);
+    DT_INST_FOREACH_CHILD(0, INTITIALIAZE_LEADER_SEQUENCES);
```

Comment:
Ditto, this adds unnecessary complexity since only a single instance of the node is supported.